direct/tpm2/read_public_test.go

```diff
@@ -51,26 +57,41 @@ func TestReadPublicKey(t *testing.T) {
 		},
 	}
 
+	// Executing the command uses reflection to pack the bytes into a 
+	// TPM2_CreatePrimary command, returns a TPM2_CreatePrimary Response.
+	// This response is also decoded so you are again working with structs 
+	// that can be found in Part 3, Commands, section 24.1.
 	rspCP, err := createPrimary.Execute(thetpm)
 	if err != nil {
 		t.Fatalf("CreatePrimary failed: %v", err)
 	}
 
+	// The TPM can only hold so much in nonvolitile memory, thus we must
+	// flush the handle after we are done using it to prevent overloading. 
+	// Again we defer the flush to after we are done using the object.
```

Comment:
It is generally good practice to `defer` the cleanup immediately after loading an object or creating an Authorization Session.

Reply:
Nice comment, also added more context from Part 1, resolved.